src/org/zaproxy/zap/utils/ZapTableExportButton.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2017 The ZAP Development Team
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.utils;
+
+import java.awt.event.ActionEvent;
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.swing.AbstractAction;
+import javax.swing.ImageIcon;
+import javax.swing.JButton;
+import javax.swing.JOptionPane;
+import javax.swing.JTable;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.zap.view.widgets.WritableFileChooser;
+
+public class ZapTableExportButton extends JButton {
+
+	private static final long serialVersionUID = 3437613469695367668L;
+
+	private static final Logger LOGGER = Logger.getLogger(ZapTableExportButton.class);
+	
+	public ZapTableExportButton(JTable table) {
+		super(Constant.messages.getString("export.button.name"));
+		super.setIcon(new ImageIcon(ZapTableExportButton.class.getResource("/resource/icon/16/115.png")));
+		super.addActionListener((new AbstractAction() {
+			private static final long serialVersionUID = 1L;
+
+			@Override
+			public void actionPerformed(ActionEvent e) {
+				WritableFileChooser chooser = new WritableFileChooser();
+				chooser.setSelectedFile(new File(Constant.messages.getString("export.button.default.filename")));
+				if (chooser.showSaveDialog(View.getSingleton().getMainFrame()) == WritableFileChooser.APPROVE_OPTION) {
+					String file = chooser.getSelectedFile().toString();
+					if (!file.endsWith(".csv")) {
+						file += ".csv";
+					}
+					try (CSVPrinter pw = new CSVPrinter(
+							Files.newBufferedWriter(chooser.getSelectedFile().toPath(), StandardCharsets.UTF_8),
+							CSVFormat.DEFAULT)) {
+						pw.printRecord(getColumnNames(table));
+						int rowCount = table.getRowCount();
+						int colCount = table.getColumnCount();
+						for (int row = 0; row < rowCount; row++) {
+							List<Object> valueOfRow = new ArrayList<Object>();
+							for (int col = 0; col < colCount; col++) {
+								valueOfRow.add(table.getValueAt(row, col).toString());

```

Comment:
This needs to be changed to handle null values (it happens with Spider panel, when a entry has no flags).